### PR TITLE
[onert] Add Shape inference for Gather

### PR DESCRIPTION
**Pull request by YiHyunJin**
This adds shape inference for Gather op

Signed-off-by: YiHyunjin <hj0412.yi@samsung.com>

### CODE REVIEW

**Comment by hyunsik-yoon** [comment]
Could you add a contributor in WG2 as a reviewer? Thanks.

File: runtime/onert/core/src/util/shapeinf/Gather.cc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  const auto rank = input.info().shape().rank();
+  const auto axis = ((op.param().axis < 0) ? rank + op.param().axis : op.param().axis);
+
+  assert(0 <= axis && axis < rank);
+
+  const auto indices_idx{op.getInputs().at(ir::operation::Gather::Input::INDICES)};
+  const auto &indices = _operands.at(indices_idx);
+  // re-sizing output shape
```
Comment (hyunsik-yoon) [inline runtime/onert/core/src/util/shapeinf/Gather.cc:70]: I wonder if `indices` can be an output of other op. If so, doesn't `indices` affect the decision of dynamic/static aspect of output?
If we want `indices` to be always const, we'd better add some `throw exception if indeces is not const` statement somewhere.

Comment (glistening) [reply runtime/onert/core/src/util/shapeinf/Gather.cc:70]: `indices` itself could be an output of other op. However, If shape of `indices` is constant, the output dimension of `Gather` could be calculated.

File: runtime/onert/core/src/util/shapeinf/Gather.cc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape GatherShapes(const ir::Shape &input_shape, const ir::Shape &indices_shape, int axis,
```
Comment (glistening) [inline runtime/onert/core/src/util/shapeinf/Gather.cc:24]: @YiHyunJin Please start with lower case.

```suggestion
ir::Shape gatherShapes(const ir::Shape &input_shape, const ir::Shape &indices_shape, int axis,
```

**Comment by juitem** [comment]
LGTM

**Comment by glistening** [comment]
@hyunsik-yoon PTAL

**Comment by intom** [comment]
LGTM

**Comment by glistening** [comment]
@intom PTAL

**Comment by hseok-oh** [comment]
Please ignore my approve

**Pull request merged**